apps/test/unit/code-studio/pd/professional_learning_landing/csPrinciplesAndDiscoveriesSectionTest.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import {mount} from 'enzyme';
+import CsPrinciplesAndDiscoveriesSection from '@cdo/apps/code-studio/pd/professional_learning_landing/csPrinciplesAndDiscoveriesSection';
+import {expect} from 'chai';
+
+describe("CS Principles and Discoveries Section", () => {
+  const generateSection = (sectionProps = []) => {
+    return mount(
+      <CsPrinciplesAndDiscoveriesSection
+        {...sectionProps}
+      />
+    );
+  };
+
+  it("should show a survey for a user that needs to take a survey", () => {
+    const section = generateSection({coursesTaught: ['CS Principles'], lastWorkshopSurveyUrl: 'url'});
+
+    expect(section.find('#cspStartSurvey')).to.have.length(1);
+    expect(section.find('#cspThanks')).to.have.length(0);
```

Comment:
I'm a little confused here. This doesn't look like a valid state. According to below, `coursesTaught` represents all courses for which this teacher is enrolled, whether or not those workshops have occurred. Surveys are not valid until the workshop has ended. Attempting to navigate to the survey before that will result in a 404.